addons/base_vat/models/res_partner.py

```diff
@@ -59,6 +59,17 @@
 class ResPartner(models.Model):
     _inherit = 'res.partner'
 
+    vat_validation_type = fields.Selection(selection_add=[('european_vat', 'European Vat')])
+
+    def get_validation_type(self):
+        self.ensure_one()
+        european_countries = []
+        if self.env.ref('base.europe'):
+            european_countries = [x.id for x in self.env.ref('base.europe').country_ids]
```

Comment:
actually, I plan to hardcode that european_countries list... I'm not even sure that the european_vat validation couldn't be used for a country outside Europe ^^